dotCMS/src/main/java/com/dotcms/publisher/bundle/business/BundleAPIImpl.java

```diff
@@ -82,7 +113,233 @@ public Bundle getBundleById(String id) throws DotDataException {
+				" and dependencies by: " + user.getUserId());
+
+		try {
+
+			Logger.info(this, "Removing audit status for a bundle: " + bundleId);
+			this.publishAuditAPI.deletePublishAuditStatus(bundleId);
+		} catch (DotPublisherException e) {
+			Logger.error(this,"Error Removing audit status for bundle: " + bundleId,e);
+			throw new DotDataException(e);
+		}
+
+		Logger.info(this, "Removing all pushed assets for a bundle: " + bundleId);
+		this.pushedAssetsAPI.deletePushedAssetsByBundleId(bundleId);
+
+		Logger.info(this, "Removing all assets from bundle: " + bundleId);
+		this.bundleFactory.deleteAllAssetsFromBundle(bundleId);
+
+		Logger.info(this, "Removing environments from bundle: " + bundleId);
+		this.bundleFactory.deleteBundleEnvironmentByBundle(bundleId);
+
+		Logger.info(this, "Removing bundle: " + bundleId);
+		this.deleteBundle(bundleId);
+	}
+
+	@WrapInTransaction
+	@Override
+	public void deleteBundleAndDependencies(final String bundleId, final User user) throws DotDataException {
+
+		final Bundle bundle = this.getBundleById(bundleId);
+		if (null != bundle) {
+
+			this.internalDeleteBundleAndDependencies(bundleId, user);
+		} else {
+			Logger.error(this,"The bundle id: " + bundleId + " does not exists");
+			throw new NotFoundInDbException("The bundle id: " + bundleId + " does not exists");
+		}
+	}
+
+	private void validateBundleDeletePermission(final User user, final String bundleId) throws DotDataException {
+
+		//This is meanwhile we make bundles permissionable, if the user is not an admin only the owner of the bundle can delete it
+		if(!userAPI.isCMSAdmin(user)){
+			if(!getBundleById(bundleId).getOwner().equals(user.getUserId())){
+				Logger.error(this,"User: " + user.getUserId() + " is not an admin or is not the owner of the bundle");
+				throw new DotDataException("User: " + user.getUserId() + " is not an admin or is not the owner of the bundle");
+			}
+		}
+	}
+
+	// no mark as a wrap in transaction, it is one trax per bundle to delete.
+	@Override
+	public Set<String> deleteBundleAndDependenciesOlderThan(final Date olderThan, final User user) throws DotDataException {
+
+		if(olderThan.after(new Date())){
+			Logger.error(this,"To avoid deleting bundles that publish in the future, the date can not be after the current date");
+			throw new IllegalArgumentException("To avoid deleting bundles that publish in the future, the date can not be after the current date");
+		}
+
+		final ImmutableSet.Builder<String> bundlesDeleted = new ImmutableSet.Builder<>();
+		final boolean isAdmin = this.userAPI.isCMSAdmin(user);
+		final int  bundleSleepCount   = Config.getIntProperty ("bundle.sleep.count",  10);  // each 10 deletes
+		final long millisBundleSleep  = Config.getLongProperty("bundle.sleep.millis", 25l); // wait 25 millis
+		final MutableInt deletedCount = new MutableInt(0);
+
+		try {
+
+			Logger.info(this, "Deleting bundles older than: " + olderThan
+					+ " by the user :" + user.getUserId());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal " by the user :" appears 4 times in this file; the first occurrence is on line 191](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4502981)